### PR TITLE
[New Resource]: aws_rds_certificate

### LINKED ISSUE #33196: [New Resource]: aws rds modify-certificates
### Description

As of this writing on August 27, 2023, the default RDS CA certificate is `rds-ca-2019`, which expires 2024-08-22. If a user creates a new RDS instance anytime prior to 2024-08-22, and they don't explicitly request a different CA certificate via the `aws_rds_instance.foo.ca_cert_identifier` attribute, then presumably their instance will get this `rds-ca-2019` certificate.

AWS has an API for `aws rds modify-certificates`, which lets you override the account-level default for new RDS instances. It would be useful if Terraform had a resource which supported this API.

If such a resource existed, a person could upgrade all their existing RDS instances to a newer and longer-duration certificate (using the `aws_rds_instance.foo.ca_cert_identifier` attribute), and then use this resource to set the account-level default for new instances so that they no longer have to worry about the old certificate ever being used again.

### Requested Resource(s) and/or Data Source(s)

* `aws_rds_account_default_ca_certficate` (or something along those lines)

### Potential Terraform Configuration

```terraform
resource "aws_rds_account_default_ca_certficate" "account_default" {
  certificate-identifier = "rds-ca-rsa2048-g1"
}
```

### References

https://docs.aws.amazon.com/cli/latest/reference/rds/modify-certificates.html

https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_ModifyCertificates.html

### Would you like to implement a fix?

No

### PR DESCRIPTION
**Pull request by obounaim**
### Description
Provides a resource to override the system-default Secure Sockets Layer/Transport Layer Security (SSL/TLS) certificate for Amazon RDS for new DB instances in the current AWS region.

### Relations

Closes #33196

### References
[AWS RDS CA](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html#UsingWithRDS.SSL.CertificateIdentifier)

### Output from Acceptance Testing

```Console
make testacc TESTS=TestAccRDSCertificate_Basic PKG=rds
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/rds/... -v -count 1 -parallel 20 -run='TestAccRDSCertificate_Basic'  -timeout 360m
=== RUN   TestAccRDSCertificate_Basic
=== PAUSE TestAccRDSCertificate_Basic
=== CONT  TestAccRDSCertificate_Basic
--- PASS: TestAccRDSCertificate_Basic (15.92s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/rds	15.981s
```

### CODE REVIEW

**Comment by obounaim** [comment]
ToDo :
- [ ] Use [DescribeCertificatesPaginator](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/rds#DescribeCertificatesPaginator)
- [ ] Finish implementing Acc Tests
- [ ] Implement `terraform import`

File: website/docs/r/rds_certificate.html.markdown [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+* `certificate_identifier` - (Required) Certificate identifier. For example, `rds-ca-rsa4096-g1`. Refer to [AWS RDS (Relational Database) Certificate Identifier](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html#UsingWithRDS.SSL.CertificateIdentifier) for more information.
+
+## Attribute Reference
+
+This resource exports no additional attributes.
+
+## Import
+
+In Terraform v1.5.0 and later, use an [`import` block](https://developer.hashicorp.com/terraform/language/import) to import the default EBS encryption state. For example:
```
Comment (jrobison-sb) [inline website/docs/r/rds_certificate.html.markdown:35]: Is "default EBS encryption state" a leftover from a copy/paste? This SSL CA certificate relates to data in transit, not data at rest on EBS.

Would it maybe be more accurate to say "import the default RDS CA certificate"?

Comment (obounaim) [reply website/docs/r/rds_certificate.html.markdown]: @jrobison-sb , thanks.
Fixed

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccRDSCertificate_serial' PKG=rds
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.4 test ./internal/service/rds/... -v -count 1 -parallel 20  -run=TestAccRDSCertificate_serial -timeout 360m
=== RUN   TestAccRDSCertificate_serial
=== PAUSE TestAccRDSCertificate_serial
=== CONT  TestAccRDSCertificate_serial
=== RUN   TestAccRDSCertificate_serial/basic
=== RUN   TestAccRDSCertificate_serial/disappears
--- PASS: TestAccRDSCertificate_serial (26.48s)
    --- PASS: TestAccRDSCertificate_serial/basic (17.61s)
    --- PASS: TestAccRDSCertificate_serial/disappears (8.86s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/rds	31.485s
```

**Comment by ewbankkit** [comment]
@obounaim Thanks for the contribution :tada: :clap:.

**Pull request merged**